posthog/warehouse/api/table.py

```diff
@@ -266,3 +272,112 @@ def refresh_schema(self, request: request.Request, *args: Any, **kwargs: Any) ->
         table.save()
 
         return response.Response(status=status.HTTP_200_OK)
+
+    @action(
+        methods=["POST"],
+        detail=False,
+        required_scopes=["warehouse_table:write"],
+        parser_classes=[parsers.MultiPartParser, parsers.FormParser],
+    )
+    def file(self, request: request.Request, *args: Any, **kwargs: Any) -> response.Response:
+        team = Team.objects.get(id=self.team_id)
+        is_warehouse_api_enabled = posthoganalytics.feature_enabled(
+            "warehouse-api",
+            str(team.organization_id),
+            groups={"organization": str(team.organization_id)},
+            group_properties={"organization": {"id": str(team.organization_id)}},
+        )
+
+        if not is_warehouse_api_enabled:
+            return response.Response(
+                status=status.HTTP_400_BAD_REQUEST, data={"message": "Warehouse API is not enabled for this endpoint"}
+            )
+
+        if "file" not in request.FILES:
+            return response.Response(status=status.HTTP_400_BAD_REQUEST, data={"message": "No file provided"})
+
+        file = request.FILES["file"]
+        table_name = request.data.get("name", file.name)
+        file_format = request.data.get("format", "CSVWithNames")
+
+        # Validate table name format
+        if not re.match(r"^[a-zA-Z_][a-zA-Z0-9_]*$", table_name) or not re.match(
+            r"^[a-zA-Z_][a-zA-Z0-9_]*$", file.name
+        ):
+            return response.Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    "message": "Table names must start with a letter or underscore and contain only alphanumeric characters or underscores."
+                },
+            )
+
+        # Validate table name
+        team_id = self.team_id
+        table = None
+        table_query = DataWarehouseTable.objects.exclude(deleted=True).filter(team_id=team_id, name=table_name)
+        if table_query.exists():
+            table = table_query.first()
+
+        if file.size > 52428800:  # 50MB in bytes
+            return response.Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={"message": f"File size exceeds maximum allowed size of 50MB"},
+            )
+
+        # Create the table record
+        try:
+            # Create credential if object storage is available
+            credential = None
+            if hasattr(settings, "AIRBYTE_BUCKET_KEY") and hasattr(settings, "AIRBYTE_BUCKET_SECRET"):
+                credential = get_or_create_datawarehouse_credential(
+                    team_id=team_id,
+                    access_key=settings.AIRBYTE_BUCKET_KEY,
+                    access_secret=settings.AIRBYTE_BUCKET_SECRET,
+                )
```

Comment:
Should we not raise if these settings dont exist? Saves the `# type: ignore` below